src/main/java/seedu/address/model/person/Name.java

```diff
@@ -18,7 +18,7 @@ public class Name implements DeepCopyable {
      * The first character of the address must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+    public static final String VALIDATION_REGEX = "^([0-9]*[a-zA-Z]+){1,}(([0-9]*[a-zA-Z ]+){1,})*$";
```

Comment:
It doesn't match quite a few cases like
```
2nd Queen Elizabeth
nd2
3ad3
```

There's another issue with catastrophic backtracking, which causes the `*` to try to match the previous group unlimited times, with no breaking point.

I have a regex that can match the required pattern
```
^(?:[a-zA-Z]*|(?:\S*[^\d\s]\d+\S*|\S*\d+[^\d\s]\S*)| )*$
```
but this will not solve catastrophic backtracking.

In any case this will always take a very long time.

I think if we can't do this properly, we'll just disable the ability to enter numbers into the name at all - so we only accept lower and uppercase alphabets and spaces and nothing else.